### ISSUE TITLE
jquery-dollar-sign-reference shold flag jQuery assignments inside LogicalExpressions

### ISSUE THREAD

**Issue by GoodForOneFare** [Shopify/eslint-plugin-shopify#2]
The following code should be flagged by the linter as requiring `var $foo`:

```js
var foo = bar || $('baz');
```

/cc @lemonmade

**Comment by lemonmade** [comment]
It's kind of a weird one, it should probably complain about both the variable assignment and the bar comparison, but I don't think we typically get into anything more complex than checking more basic assignments.

In any case, I agree that we should at least flag the assignment 👍
